shared/src/commonMain/kotlin/com.macaosoftware.sdui.app/marketplace/amadeus/auth/AuthViewModel.kt

```diff
@@ -26,12 +37,66 @@ class AuthViewModel(
 
     }
 
-    fun login() {
+    suspend fun login(email: String, password: String) {
         authPlugin.login(
-            LoginRequest("pablo@gmail.com", "pass123") {
-                println("Login Result: $it")
+            LoginRequest(email, password) { result ->
+                handleLoginResult(result)
+            }
+        )
+    }
+
+    suspend fun signup(email: String, password: String, username: String, phoneNo: String) {
+        authPlugin.signup(
+            SignupRequest(email, password, username, phoneNo) { result ->
+                handleSignupResult(result)
             }
         )
     }
 
+    private fun handleLoginResult(result: MacaoResult<MacaoUser>) {
+        if (result != null) {
+            if (result is MacaoResult.Success) {
+                val macaoUser = result.value
+                println("Login Successful: $macaoUser")
+            } else if (result is MacaoResult.Error) {
+                val loginError = result.error
+                println("Login Failed: $loginError")
+            }
+        } else {
+            println("Login result is null")
+        }
+    }
+
+    private fun handleSignupResult(result: MacaoResult<MacaoUser>) {
+        if (result != null) {
+            if (result is MacaoResult.Success) {
+                val macaoUser = result.value
+                println("Signup Successful: $macaoUser ")
+            } else if (result is MacaoResult.Error) {
+                val signupError = result.error
+                println("Signup Failed: $signupError")
+            }
+        } else {
+            println("Signup result is null")
+        }
+    }
+
+    suspend fun storeData(user: User) {
+        val firebaseUser = Firebase.auth.currentUser?.uid
```

Comment:
This code should be in a datastore plugin not directly here. Macao shared module shouldn't access Firebase.